io/io/v7/src/TFile.cxx

```diff
@@ -124,75 +122,78 @@ static std::mutex& GetCacheDirMutex() {
   return sMutex;
 }
 
-static std::unique_ptr<ROOT::Experimental::Internal::TFileStorageInterface>
-OpenV6TFile(std::string_view name, const char* mode,
-            const ROOT::Experimental::TFile::Options_t& opts) {
-  // Set and re-set the cache dir.
-  // FIXME: do not modify a static here, pass this to the underlying Open.
-  struct SetCacheDirRAII_t {
-    std::string fOldCacheDir;
-    std::lock_guard<std::mutex> fLock;
-
-    SetCacheDirRAII_t(bool need): fLock(GetCacheDirMutex()) {
-      if (need)
-        fOldCacheDir = TFile::GetCacheFileDir();
-    }
+static std::unique_ptr<ROOT::Experimental::Internal::TFileStorageInterface> OpenV6TFile(
+   const std::string &name, const char *mode, const ROOT::Experimental::TFile::Options_t &opts)
+{
+   // Set and re-set the cache dir.
```

Comment:
I applied clang-format to each commit. If you insist I can separate the clang-format changes, but why not commit in the right format directly?

As for the other comments, I will address them and push a new version soon.